pkg/network/config_generator.go

```diff
@@ -0,0 +1,29 @@
+package network
+
+import (
+	"fmt"
+	"io"
+	"os/exec"
+)
+
+type ConfigGenerator struct{}
+
+func (ConfigGenerator) GenerateNetworkConfig(configDir, outputDir string, outputWriter io.Writer) error {
+	cmd := generateCommand(configDir, outputDir, outputWriter)
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("running generate command: %w", err)
+	}
+
+	return nil
+}
+
+func generateCommand(configDir, outputDir string, output io.Writer) *exec.Cmd {
+	cmd := exec.Command("nmc", "generate",
+		"--config-dir", configDir,
+		"--output-dir", outputDir)
+
+	cmd.Stdout = output
+	cmd.Stderr = output
```

Comment:
It will definitely work for xorriso. Not sure what the issue you mention has been at the time.

Reply:
Now that I think about it, this is the solution I went with (except to a log file instead of stdout/stderr). Originally I tried to capture the output using `cmd.Output()` and that didn't behave the way I expected.

Reply:
You were probably ignoring the output result when there is an error. This is one of the few occasions where a function can return both non-nil result and non-nil error. But then again, if you want to also capture stderr then `CombinedOutput()` is better.